pkg/jobs/canary/sync_upstream.go

```diff
@@ -0,0 +1,91 @@
+package canary
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/flanksource/canary-checker/api/context"
+	v1 "github.com/flanksource/canary-checker/api/v1"
+	"github.com/flanksource/canary-checker/pkg/db"
+	"github.com/flanksource/commons/logger"
+	"github.com/flanksource/duty/models"
+	"github.com/flanksource/duty/upstream"
+	"gorm.io/gorm/clause"
+)
+
+var UpstreamConf upstream.UpstreamConfig
+
+var tablesToReconcile = []string{
+	"checks",
+	"check_statuses",
+}
+
+// PushCanaryResultsToUpstream coordinates with upstream and pushes any resource
+// that are missing on the upstream.
+func PushCanaryResultsToUpstream() {
+	ctx := context.New(nil, nil, db.Gorm, v1.Canary{})
+
+	jobHistory := models.NewJobHistory("PushCanaryResultsToUpstream", "Canary", "")
+	_ = db.PersistJobHistory(jobHistory.Start())
+	defer func() { _ = db.PersistJobHistory(jobHistory.End()) }()
+
+	reconciler := upstream.NewUpstreamReconciler(UpstreamConf, 5)
+	for _, table := range tablesToReconcile {
+		if err := reconciler.Sync(ctx, table); err != nil {
+			jobHistory.AddError(err.Error())
+			logger.Errorf("failed to sync table %s: %v", table, err)
+		} else {
+			jobHistory.IncrSuccess()
+		}
+	}
+}
+
+func Pull() {
```

Comment:
rename to something more descriptive, like `PullUpstreamCanary` etc.
`Pull` in canary package is very vague